### PR TITLE
adding email service & instialltion of mail gun

### PR DESCRIPTION
**Pull request by Just-Jaden**
### Trello Link
[Ticket](https://trello.com/c/oQ9eedlu)

# Description

installed mailgun to then create an `emailService`

## Usage
first, import this service: 
```javascript 
import { sendEmail, sendHtmlEmail } from './src/services/emailService.js';
```
then use it to send an email with text 
```javascript 
sendEmail({
  to: 'exampleEmail@example.com',
  subject: 'YOUR_SUBJECT_HERE',
  text: 'YOUR_TEXT_HERE',
});
```
or to send an email with Html 
```javascript
sendEmail({
  to: 'exampleEmail@example.com',
  subject: 'YOUR_SUBJECT_HERE',
  Html: 'YOUR_HTML_HERE',
});
```
## Type of change

- [ ] Bug fix
- [x ] New feature
- [ ] Infrastructure

### How Has This Been Tested?

 - ran locally

 ### New Environment Variables (if any)

- MAILGUN_API_KEY
- MAILGUN_DOMAIN

### Checklist:

- [x] I have performed a self-review of my own code

### CODE REVIEW

File: src/services/emailService.js [hunk 1]
```diff
@@ -0,0 +1,89 @@
+import _ from 'lodash-es';
+import validator from 'validator';
+import dotenv from 'dotenv';
+import formData from 'form-data';
+import Mailgun from 'mailgun.js';
+import Logger from '../tools/logger.js';
+import { EmailServiceError } from '../constants/commonErrors.js';
+
+dotenv.config();
```
[the remark this answers is not in the document]
Comment (Just-Jaden) [reply src/services/emailService.js:9]: should I replace the current line in `server.js` that has the following, 
```javascript 
import Logger from './src/tools/logger.js';
```
with,
```javascript
import Logger from '../tools/logger.js';
```  
or add it ontop of it?

File: src/services/emailService.js [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+  }
+}
+
+export async function sendEmail({ to, subject, text }) {
+  validateSubject(subject);
+  validateText(text);
+  validateRecipient(to);
+  setupClient();
+  await EMAIL_CLIENT.messages.create(process.env.MAILGUN_DOMAIN, {
+    from: `Hello@${process.env.MAILGUN_DOMAIN}`,
```
Comment (KevinHuang0126) [inline src/services/emailService.js:63]: Is `process.env.MAILGUN_DOMAIN` just going to be our URL for the website?

File: src/services/emailService.js [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+    to: to,
+    subject: subject,
+    text: text,
+  });
+}
+
+export async function sendHtmlEmail({ to, subject, html }) {
```
Comment (KevinHuang0126) [inline src/services/emailService.js:70]: What's the difference in functionality and purpose of `sendEmail` and `sendHtmlEmail`?

**Review by KevinHuang0126 — APPROVED** [review]
looks good!! 👍👍

File: src/services/emailService.js (hunk 2 above)
Comment (Just-Jaden) [reply src/services/emailService.js:63]: Yes though we dont have one yet so we are using the domain of the default mailgun application

File: src/services/emailService.js (hunk 3 above)
Comment (Just-Jaden) [reply src/services/emailService.js:70]: `sendEmail` and `sendHtmlEmail` are practically the same requiring the same parameters "subject" & "recipent" the only thing that differs between them is the way that text is sent one is just a string with a text message (`sendEmail`) hence the variable name text, and the other(`sendHtmlEmail`) is a line or several of HTML and can use html tags such as `"<b>"` and or `"<strong>"` <b>to change the apperace and weight of said text!</b>

Comment (KevinHuang0126) [reply src/services/emailService.js:70]: I see! Thanks!

**Pull request merged**